src/handlers/records-delete.ts

```diff
@@ -86,12 +90,15 @@ export class RecordsDeleteHandler implements MethodHandler {
     }
 
     const recordsWrite = await RecordsWrite.getInitialWrite(existingMessages);
-    const indexes = RecordsDeleteHandler.constructIndexes(recordsDelete, recordsWrite);
-    await this.messageStore.put(tenant, message, indexes);
-
+    const published = (newestExistingMessage as RecordsWriteMessage).descriptor.published;
```

Comment:
For my (and future reader's) understanding: can you add a comment note as to why it is critical to have the latest `published` in the index? I vague remember you mention it has something to do with selective sync needing to detect the delete or something but not 100% sure.

Though, on the related note, should we use the latest value of other mutable indexes? Such as `dataSize` and `dataCid`?

Reply:
Sanity follow up: the `published` index is no longer need for scenario you were after?